Source/WebCore/html/parser/HTMLConstructionSite.cpp

```diff
@@ -594,21 +625,25 @@ void HTMLConstructionSite::insertTextNode(const String& characters, WhitespaceMo
     // for performance, see <https://bugs.webkit.org/show_bug.cgi?id=55898>.
 
     RefPtr<Node> previousChild = task.nextChild ? task.nextChild->previousSibling() : task.parent->lastChild();
-    if (is<Text>(previousChild)) {
-        // FIXME: We're only supposed to append to this text node if it
-        // was the last text node inserted by the parser.
-        currentPosition = downcast<Text>(*previousChild).parserAppendData(characters, 0, lengthLimit);
+    if (auto* previousTextChild = dynamicDowncast<Text>(previousChild.get())) {
```

Comment:
Updating this line to:
`if (auto* previousTextChild = dynamicDowncast<Text>(previousChild.get()); previousTextChild && previousTextChild->length() < lengthLimit) {`

To protect against underflows.